modules/internal/ChapelArray.chpl

```diff
@@ -610,6 +616,46 @@ module ChapelArray {
     return chpl__distributed(d, chpl__buildDomainExpr((...ranges)));
   }
 
+  //
+  // Array-view utility functions
+  //
+  proc chpl__isArrayView(arr) param {
+    use Reflection;
+    param isSlice = if canResolveMethod(arr, "isSliceArrayView") then arr.isSliceArrayView() else false;
+    param isRankChange = if canResolveMethod(arr, "isRankChangeArrayView") then arr.isRankChangeArrayView() else false;
+    param isReindex = if canResolveMethod(arr, "isReindexArrayView") then arr.isReindexArrayView() else false;
+
+    return isSlice || isRankChange || isReindex;
+  }
+
+  //
+  // Returns a domain with a rank equivalent to chpl__getActualArray(arr).rank.
+  // This domain is no larger than the innermost array's domain. It represents
+  // the 'active' indices that the top-level ArrayView works with. For example:
+  //
+  //   var A : [1..10, 1..10];
+  //   var B => A[1, 1..10];
+  //   writeln(chpl__getViewDom(B)); // {1..1, 1..10}
+  //
+  proc chpl__getViewDom(arr: []) {
+    if chpl__isArrayView(arr._value) then return arr._value._getViewDom();
```

Comment:
I wonder if this should be similiarly flexible to getActualArray so that it can accept an array class in addition to an array?